casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -87,9 +87,11 @@ object BlockAPI {
                 result <- maybeBlock match {
                            case err: NoBlock =>
                              Metrics[F]
-                               .incrementCounter("propose-failed") >> s"Error while creating block: $err"
-                               .asLeft[DeployServiceResponse]
-                               .pure[F]
+                               .incrementCounter("propose-failed") >>
+                               Log[F].error(s"Error while creating block: $err") >>
```

Comment:
I don't think this should be logged at the error level. I usually interpret error-level log entries as "something went terribly wrong". The errors we are talking about here are not fatal (no new deploys, synchrony constraint, read only mode etc) so I think warning suits better here.

Reply:
You are maybe right. I'll change it.

Reply:
Now we won't raise here any exceptions anymore.